### ISSUE TITLE
All critical checkpoint passed, but still get error to execute the command. 

### ISSUE THREAD

**Issue by chenzhendong** [aws-containers/amazon-ecs-exec-checker#43]
Got everything passed but still cannot run the command due to error, is there any other checkpoint need to be added to the script?

% aws ecs execute-command  \
    --region $AWS_REGION \
    --cluster rd-xxx-xxxx-ecs-cluster  \
    --task 9496e2baead84a7d8116e989c2bca473 \
    --container xxx-xxxxx \
    --command /bin/bash \
    --interactive\
    --profile dev

The Session Manager plugin was installed successfully. Use the AWS CLI to start a session.
An error occurred (TargetNotConnectedException) when calling the ExecuteCommand operation: The execute command failed due to an internal error. Try again later.

The script result is as below

% bash <( curl -Ls https://raw.githubusercontent.com/aws-containers/amazon-ecs-exec-checker/main/check-ecs-exec.sh ) rd-dlas-op1service-ecs-cluster 9496e2baead84a7d9116e903c2bca473
-------------------------------------------------------------
Prerequisites for check-ecs-exec.sh v0.7
-------------------------------------------------------------
  jq      | OK (/usr/local/bin/jq)
  AWS CLI | OK (/usr/local/bin/aws)

-------------------------------------------------------------
Prerequisites for the AWS CLI to use ECS Exec
-------------------------------------------------------------
  AWS CLI Version        | OK (aws-cli/2.2.32 Python/3.8.8 Darwin/20.6.0 exe/x86_64 prompt/off)
  Session Manager Plugin | OK (1.2.245.0)

-------------------------------------------------------------
Checks on ECS task and other resources
-------------------------------------------------------------
Region : us-east-1
Cluster: rd-xxx-xxxx-ecs-cluster
Task   : 9496e2baead84a7d8916e903c2bca473
-------------------------------------------------------------
  Cluster Configuration  | Audit Logging Not Configured
  Can I ExecuteCommand?  | arn:aws:iam::0000000:role/US-_AWS000000_Developer_2
     ecs:ExecuteCommand: allowed
     ssm:StartSession denied?: allowed
  Task Status            | RUNNING
  Launch Type            | Fargate
  Platform Version       | 1.4.0
  Exec Enabled for Task  | OK
  Container-Level Checks | 
    ----------
      Managed Agent Status
    ----------
         1. STOPPED (Reason: null) for "xxx-xxxproxy" - LastStartedAt: null
         2. RUNNING for "xxx-xxxservice"
    ----------
      Init Process Enabled (rd-dlas-op1service:72)
    ----------
         1. Enabled - "xxx-xxxservice"
         2. Enabled - "xxx-xxxproxy"
    ----------
      Read-Only Root Filesystem (rd-dlas-op1service:72)
    ----------
         1. Disabled - "xxx-xxxservice"
         2. Disabled - "xxx-xxxproxy"
  Task Role Permissions  | arn:aws:iam::xxxxxxxx:role/ADD-rd-xxx-xxx-ECSTask
     ssmmessages:CreateControlChannel: allowed
     ssmmessages:CreateDataChannel: allowed
     ssmmessages:OpenControlChannel: allowed
     ssmmessages:OpenDataChannel: allowed
  VPC Endpoints          | 
    Found existing endpoints for vpc-039e987a:
      - com.amazonaws.us-east-1.s3
      - com.amazonaws.us-east-1.execute-api
      - com.amazonaws.us-east-1.ecr.api
      - com.amazonaws.us-east-1.ecr.dkr
      - com.amazonaws.us-east-1.cloudformation
      - com.amazonaws.us-east-1.ecs
      - com.amazonaws.us-east-1.ecs-agent
      - com.amazonaws.us-east-1.ecs-telemetry
      - com.amazonaws.us-east-1.logs
      - com.amazonaws.us-east-1.sqs
      - com.amazonaws.vpce.us-east-1.vpce-svc-xxxx
      - com.amazonaws.vpce.us-east-1.vpce-svc-xxxxx
      - com.amazonaws.us-east-1.dynamodb
      - com.amazonaws.us-east-1.monitoring
      - com.amazonaws.us-east-1.ssm
      - com.amazonaws.us-east-1.ec2
      - com.amazonaws.us-east-1.ec2messages
      - com.amazonaws.us-east-1.ssm
      - com.amazonaws.us-east-1.ec2messages
      - com.amazonaws.us-east-1.ssmmessages
      - com.amazonaws.us-east-1.kms
      - com.amazonaws.us-east-1.sns
      - com.amazonaws.us-east-1.glue
      - com.amazonaws.us-east-1.execute-api
      - com.amazonaws.us-east-1.secretsmanager
      - com.amazonaws.vpce.us-east-1.vpce-svc-xxxxxx
      - com.amazonaws.us-east-1.kinesis-firehose
      - com.amazonaws.us-east-1.textract
      - com.amazonaws.us-east-1.states
      - com.amazonaws.us-east-1.autoscaling
      - com.amazonaws.us-east-1.email-smtp

**Comment by toricls** [comment]
Hi @chenzhendong, thank you for reaching out to us and sorry for taking time to respond.

The error "TargetNotConnectedException" you saw generally means that the ssm-agent inside your container, which is bind mounted by ECS, is not able to establish a connection to the SSM service.

What I found from the script result you provided is that the managed agent (= ssm-agent) is not actually running. So in this case you cannot connect at least to the container `xxx-xxxproxy`.

```
----------
Managed Agent Status
----------
1. STOPPED (Reason: null) for "xxx-xxxproxy" - LastStartedAt: null
```

ECS regularly tries to recover the agent status by restarting the agent, so I suspect that there might be something unexpected going on in the AWS services side if this is still happening to you. If so it would be better for you to create a support ticket to AWS support so that they can investigate much more than here on GitHub to solve the problem.

**Comment by chenzhendong** [comment]
Thanks for the reply.  Is it possible I attach to another containe with the managed agent? I I tried to connect the container  for "xxx-xxxservice", got the same error

**Comment by toricls** [comment]
Yes, you should be able to exec into the "xxx-xxxservice" container in this case. But make sure that the script result is a kind of snapshot, and the actual status of the managed agent could be changed.
So it would be useful to run the script again and see what you get few mins after you see the error.
